scripts/autotools/libmongoc/CheckCompression.m4

```diff
@@ -59,10 +80,14 @@ if test "$found_snappy" = "yes" -o "$found_zlib" = "yes" -o "$bundled_zlib" = "y
   else
     AC_SUBST(MONGOC_ENABLE_COMPRESSION_ZLIB, 0)
   fi
+  if test "$found_zstd" = "yes"; then
+    AC_SUBST(MONGOC_ENABLE_COMPRESSION_ZSTD, 1)
```

Comment:
While we don't have any build flags for compression (we just link whatever is available) nor do we have Evergreen environments w/ and w/o these libs, I did both detection branches (`pkg-config` and the fallback) locally.

On Ubuntu, the relevant packages are `libzstd-dev` and `libzstd1`. I first ran `configure` and verified that the `pkg-config` branch picked up "libzstd" and added `-lzstd` to our link flags. Relevant `configure` output is simply:

```
checking for PHP_MONGODB_ZSTD... yes
```

I then modified the M4 script to fail the `pkg-config` check by adding a typo in the "libzstd" package name to verify that the `PHP_CHECK_LIBRARY` and `AC_CHECK_HEADER` fallback successfully added the same link flag. The `configure` output in this case changes to:

```
checking for PHP_MONGODB_ZSTD... no
checking for ZSTD_compress in -lzstd... yes
checking zstd.h usability... yes
checking zstd.h presence... yes
checking for zstd.h... yes
```